crates/ty_python_semantic/resources/mdtest/enums.md

```diff
@@ -377,6 +413,75 @@ for color in Color:
 reveal_type(list(Color))  # revealed: list[Unknown]
 ```
 
+## Methods / non-member attributes
+
+Methods and non-member attributes defined in the enum class can be accessed on enum members:
+
+```py
+from enum import Enum
+
+class Answer(Enum):
+    YES = 1
+    NO = 2
+
+    def is_yes(self) -> bool:
+        return self == Answer.YES
+    constant: int = 1
+
+reveal_type(Answer.YES.is_yes())  # revealed: bool
+reveal_type(Answer.YES.constant)  # revealed: int
+
+class MyEnum(Enum):
+    def some_method(self) -> None:
+        pass
+
+class MyAnswer(MyEnum):
+    YES = 1
+    NO = 2
+
+reveal_type(MyAnswer.YES.some_method())  # revealed: None
+```
+
+## Accessing enum members from `type[…]`
+
+```py
+from enum import Enum
+
+class Answer(Enum):
+    YES = 1
+    NO = 2
+
+def _(answer: type[Answer]) -> None:
```

Comment:
do we internally simplify `type[Answer]` to a class-literal type, the same as we do for `type[C]` where `C` is explicitly decorated with `@final`? Is it worth adding a test for that?

Do we have any tests for enum classes that are explicitly decorated with `@final`? Even though it's unnecessary, it should probably be allowed

Reply:
I fixed this by properly modeling the implicit final property of enum classes with members. This also resolved a few other TODOs (intersections collapse to Never, emit a diagnostic if you attempt to subclass an enum).